### ISSUE TITLE
RHL Babel patch breaks TypeScript declarations using `declare`

### ISSUE THREAD

**Issue by sciyoshi** [gaearon/react-hot-loader#1290]
### Description

When react-hot-loader's babel plugin processes a TypeScript file containing a `declare` statement, it leads to a runtime error.

### Expected behavior

react-hot-loader should not attempt to register a binding for a TS declaration that does not output a corresponding JS declaration.

### Environment

node: v10.15.0
npm: 6.4.1
react-hot-loader: 4.12.3
@babel/plugin-transform-typescript: 7.4.5
typescript: 3.5.2
operating system: Ubuntu 19.04

### Reproducible Demo

Set up a minimal create-react-app, and add the following line to any file:
```
declare const noValue: unique symbol;
```
RHL's babel plugin will register a binding in the output:
```
reactHotLoader.register(noValue, "noValue", "/home/sciyoshi/workspace/web/hotload/my-app/src/App.tsx");
```
However, the transpiled TypeScript has no corresponding declaration for `noValue`, leading to a runtime error: `ReferenceError: noValue is not defined`

I have a fix that works locally, but I'm not familiar enough with Babel's AST to be confident that it's correct:

https://github.com/gaearon/react-hot-loader/compare/master...sciyoshi:patch-1#diff-66ea0cdc3989360adbd4955babda95e0

**Comment by theKashey** [comment]
Released as 4.12.4, thank you

**Comment by sciyoshi** [comment]
Thank you for the quick fix!
